### PR TITLE
REGALE library

### PR DESCRIPTION
**Pull request by danielecesarini**
This PR contains a package of the REGALE library

### CODE REVIEW

File: var/spack/repos/builtin/packages/regale/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    variant("examples", default=False, description="Build examples")
+    variant("shared", default=True, description="Build shared libraries")
+
+    def cmake_args(self):
+        args = []
+        args += [self.define_from_variant("REGALE_EXAMPLES", "examples")]
+        args += [self.define_from_variant("BUILD_SHARED_LIBS", "shared")]
+        # Remove empty strings
+        args = [arg for arg in args if arg]
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/regale/package.py]: Why is this needed? Can't we just:
```suggestion
```
?

File: var/spack/repos/builtin/packages/regale/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    depends_on("eprosima-fastdds")
+
+    variant("examples", default=False, description="Build examples")
+    variant("shared", default=True, description="Build shared libraries")
+
+    def cmake_args(self):
+        args = []
+        args += [self.define_from_variant("REGALE_EXAMPLES", "examples")]
+        args += [self.define_from_variant("BUILD_SHARED_LIBS", "shared")]
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/regale/package.py]: We can return a list literal for better readability

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/regale/package.py (hunk 2 above)
Comment (danielecesarini) [reply var/spack/repos/builtin/packages/regale/package.py]: done

**Comment by danielecesarini** [comment]
@alalazo if you are able to merge this PR by the end of today would be great (if the code is ok). Thanks

**Review by alalazo — APPROVED** [review]

**Pull request merged**